# 03 - Statistical Analysis
Each test below states its own null/alternative hypothesis, statistic and business interpretation - see `src/statistical_analysis.py`.

In [1]:
import sys; sys.path.insert(0, '../src')
from data_preprocessing import run as preprocess
import statistical_analysis as sa

df, _ = preprocess('../data/coffee_shop_sales.csv')
results = sa.run(df)
results

{'weekday_weekend_ttest': {'test': "Welch's two-sample t-test",
  'h0': 'Mean transaction revenue is equal on weekends and weekdays.',
  'h1': 'Mean transaction revenue differs between weekends and weekdays.',
  'n_weekday': 107510,
  'n_weekend': 41606,
  'mean_weekday': np.float64(4.6841),
  'mean_weekend': np.float64(4.6922),
  't_stat': np.float64(0.3278),
  'p_value': np.float64(0.7430371074060356),
  'significant_at_0.05': False},
 'store_anova': {'test': 'One-way ANOVA',
  'h0': 'Mean transaction revenue is equal across all three store locations.',
  'h1': "At least one store's mean transaction revenue differs.",
  'f_stat': np.float64(36.0889),
  'p_value': np.float64(2.1407919938542047e-16),
  'group_means': {'Astoria': 4.5899,
   "Hell's Kitchen": 4.6617,
   'Lower Manhattan': 4.8147},
  'significant_at_0.05': True},
 'category_store_chi_square': {'test': 'Chi-square test of independence',
  'h0': 'Product-category mix purchased is independent of store location.',
  'h1': 'Pr

**Interpretation**

1. **Weekday vs weekend t-test** - p = 0.74, not significant. Average spend *per transaction* does not differ by weekend/weekday; any weekend revenue difference would come from transaction *volume*, not basket size.
2. **Store ANOVA** - p < 0.001, significant. Lower Manhattan has a meaningfully higher average transaction value than Astoria; worth a targeted look at Astoria's menu mix or price points.
3. **Chi-square (category x store)** - p < 0.001, significant. Product mix genuinely differs by neighborhood, which supports store-specific (not one-size-fits-all) assortment and promo planning.
4. **Price-quantity correlation** - weak negative correlation (r ≈ -0.12) that is statistically significant only because n is large (~149K rows); the *practical* effect size is small, a reminder that statistical and business significance are not the same thing.